docs/resources/security_monitoring_default_rule.md

```diff
@@ -0,0 +1,51 @@
+page_title: "datadog_security_monitoring_default_rule"
+---
+
+# datadog_security_monitoring_default_rule Resource
+
+Provides a Datadog [Security Monitoring Rule API](https://docs.datadoghq.com/api/v2/security-monitoring/) resource for default rules.
+
+## Example Usage
+
+Enable a default rule and configure it's notifications.
+```hcl
+resource "datadog_security_monitoring_default_rule" "adefaultrule" {
+    rule_id = "ojo-qef-3g3"
+    enabled = true
+
+    # Change the notifications for the high case
+    case {
+        status = "high"
+        notifications = ["@me"]
+    }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+- `rule_id`: (Required) The id of a default rule.
+- `enabled`: (Optional) Enable the default rule.
+- `disabled`: (Optional) Disable the default rule.
+- `case`: (Optional) Change the notifications of a case.
+    - `status`: Severity of the case.
+    - `notifications`: Notification targets for the case.
+
+## Configuring many rules
+
+It is possible to configure many default rules using the `datadog_security_monitoring_rules` datasource.
+
+```hcl
+# List all default rules with tag "security:attack"
+datasource "datadog_security_rules" "attack_default_rules" {
+    default_only_filter = true
+    tags_filter = ["security:attack"]
+}
+
+# Create a resource for each default rule id and enable the rule
+resource "datadog_security_monitoring_default_rule" "attack_rule" {
+    for_each = toset(data.datadog_security_monitoring_rules.rules.rule_ids)
+    rule_id = each.key
+    enabled = true
+}
```

Comment:
IIUC this is the motivation behind `datadog_security_rules` datasource. It is a peculiar use of datasources AFAICT. For instance, how does it behave if you create datasources that overlaps? Say you have `attack_default_rules` and `specific_framework` rules. A rule could belong to both, and would result in multiple `datadog_security_monitoring_default_rule`. My guess is that in this case, last write wins. but with lots of subtleties. Are we comfortable to recommend such usage?

I would have recommended dealing with default rules with a prior import.

Reply:
IIUC your edge case is something like this ?
```terraform
resource "datadog_security_monitoring_default_rule" "rule1" {
    rule_id = "SAMEID"
    enabled = true
}

resource "datadog_security_monitoring_default_rule" "rule2" {
    rule_id = "SAMEID"
    enabled = false
}
```

I'm actually unsure what whould Terraform do here, It will probably conflict on the resource ID depending on how you build it.

Reply:
I am asking @NicolasLevaique what he thinks about this as well

Reply:
Also I just thought of a resource in the AWS provider that has a similar behavior, with no real `create/delete`.
So at least this behavior exist.

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/default_vpc
https://github.com/hashicorp/terraform-provider-aws/blob/master/aws/resource_aws_default_vpc.go

Might be worth checking if they handle this edge case with two resource with different params.

Reply:
Also maybe a bit late but there's `setintersection`, `setunion` and `setsubtract` functions that could have helped user with that issue.